odoo/fields.py

```diff
@@ -565,7 +565,7 @@ def traverse_related(self, record):
         """ Traverse the fields of the related field `self` except for the last
         one, and return it as a pair `(last_record, last_field)`. """
         for name in self.related[:-1]:
-            record = record[name][:1]
+            record = record[name][:1].with_prefetch(record._prefetch)
```

Comment:
:+1: That subtle slice (for traversing x2m) had the unexpected side-effect of dropping the prefetch set.
This is consistent with the spirit of 8bc7d8565c3e4f954c479e3b51b863e4c03dc582 and the prefetch propagation for relationships ([x2m](https://github.com/odoo/odoo/blob/89360f9d9bcbf8a83d0870d305aba6c863a4e935/odoo/fields.py#L2015) and [m2o](https://github.com/odoo/odoo/blob/89360f9d9bcbf8a83d0870d305aba6c863a4e935/odoo/fields.py#L1913))